localstack/services/kinesis/kinesis_listener.py

```diff
@@ -150,7 +151,16 @@ def return_response(self, method, path, data, headers, response):
                 if sdk_v2:
                     record['ApproximateArrivalTimestamp'] = int(record['ApproximateArrivalTimestamp'])
                 if not isinstance(record['Data'], str):
-                    record['Data'] = bytearray(record['Data']['data']).decode('utf-8')
+                    # Remove double quotes from data written as bytes
+                    # https://github.com/localstack/localstack/issues/3588
+                    tmp = bytearray(record['Data']['data'])
+                    tmp = base64.b64encode(tmp).decode('utf-8')
+                    tmp = base64.b64decode(tmp).decode('utf-8').strip('"')
+                    record['Data'] = bytearray(tmp, 'utf-8')
+                elif isinstance(record['Data'], str):
+                    tmp = base64.b64decode(record['Data']).decode('utf-8').strip('"')
```

Comment:
nit: we usually avoid using `.decode(..)` directly, and instead use `to_str(..)`/`to_bytes(..)` from `utils/common.py` for `str`<->`bytes` conversions.

Same issue regarding stripping of `"` as above - would something like this work?
```
tmp = base64.b64decode(record['Data'])
if len(tmp) >= 2 and tmp[0] == tmp[-1] == b'"':
    tmp = tmp[1:-1]
record['Data'] = to_str(base64.b64encode(tmp))
```